CHANGELOG.md

```diff
@@ -1,3 +1,50 @@
+## v0.7.0
+- Moved all environment variables from `POWERLEVEL9K_` to `P9K_`
+- `utilities.zsh`
+  - Added OS detection logic to allow use of `$OS` in segments.
+  - Added OS version detection logic for Linux to allow use of `$OS_ID` in segments.
+  - Added Terminal detection logic to allow use of `$TERMINAL` in segments.
+  - Added function `updateEnvironmentVars()` to automatically change user variables for backward compatibility. NOTE: There may be minor issues with arrays, so this still needs review.
+- `icons.zsh`
+  - Only core icons are registered by default.
+  - Each segment is responsible to register its own icons.
+- Moved all segments to their own files under `segments`.
+  - Segments now use `serializeSegment()` instead of `$1_prompt_segment()`.
+  - Only segments in use are dynamically loaded (including their icons).
+  - The autoloader checks if a file exists before calling it to catch any errors. For `custom_prompt` segments, it checks if it is a variable/definition before checking for a file.
+  - Custom segment files have to be placed in `segments/custom` directory, and have to follow the naming convention.
+    - Segment file should be named `segment_name.p9k`.
+    - Segment definition should be `prompt_custom_segment_name()`.
+- Moved prompt generators to their own files under `generator`.
+  - `default.p9k` is the default (current) generator.
+  - `zsh-async` has been included but is still experimental. To use, set `POWERLEVEL9K_GENERATOR="zsh-async"` in `powerlevel9k.zsh`.
+- Segments and generators use the `*.p9k` extention to ensure that they don't conflict with similarly names `*.zsh` scripts.
+- Segment updates
+  - `vi_mode.p9k`
+    - Rewrote to support both `default` and `zsh-async` generators.
+    - Added support to change cursor shape.
+    - To enable cursor shape change, set `POWERLEVEL9K_CURSOR_SHAPE=true`. On supported terminals, cursor shape can be `box` [], `hbar` _ or `vbar` |.
+    - Changed `zle reset-prompt` to `zle .reset-prompt`.
```

Comment:
`zle .reset-prompt` instead of `zle reset-prompt` — bypass the `zsh-syntax-highlighting` wrapper (see `_zsh_highlight_bind_widgets`).

Use the builtin dot-prefixed version of the widget instead, if the plugin only wants the default behavior to occur.

Reply:
And that is what I would expect there "to bypass zsh-syntax-highlighting wrapper". Otherwise the user that just reads that part would have no follow-up.

Btw. do you (or anybody) know if that bypasses _any_ wrapper, or is it specifically for `zsh-syntax-highlighting`?